File: ios/vendored/sdk44/@stripe/stripe-react-native/ios/CardFieldManager.m
#import <Foundation/Foundation.h>
#import <ABI44_0_0React/ABI44_0_0RCTBridgeModule.h>
#import <ABI44_0_0React/ABI44_0_0RCTViewManager.h>

@interface ABI44_0_0RCT_EXTERN_REMAP_MODULE(CardFieldManager, ABI44_0_0CardFieldManager, ABI44_0_0RCTViewManager)
ABI44_0_0RCT_EXPORT_VIEW_PROPERTY(postalCodeEnabled, BOOL)
ABI44_0_0RCT_EXPORT_VIEW_PROPERTY(onCardChange, ABI44_0_0RCTDirectEventBlock)
ABI44_0_0RCT_EXPORT_VIEW_PROPERTY(onFocusChange, ABI44_0_0RCTDirectEventBlock)
ABI44_0_0RCT_EXPORT_VIEW_PROPERTY(cardStyle, NSDictionary)
ABI44_0_0RCT_EXPORT_VIEW_PROPERTY(placeholder, NSDictionary)
ABI44_0_0RCT_EXPORT_VIEW_PROPERTY(autofocus, BOOL)
ABI44_0_0RCT_EXPORT_VIEW_PROPERTY(dangerouslyGetFullCardDetails, BOOL)
ABI44_0_0RCT_EXTERN_METHOD(focus:(nonnull NSNumber*) ABI44_0_0ReactTag)
ABI44_0_0RCT_EXTERN_METHOD(blur:(nonnull NSNumber*) ABI44_0_0ReactTag)
ABI44_0_0RCT_EXTERN_METHOD(clear:(nonnull NSNumber*) ABI44_0_0ReactTag)
@end

File: ios/vendored/sdk44/@stripe/stripe-react-native/ios/CardFieldManager.swift
import Foundation

@objc(ABI44_0_0CardFieldManager)
class CardFieldManager: ABI44_0_0RCTViewManager {
    override func view() -> UIView! {
        let cardField = CardFieldView()
        let stripeSdk = bridge.module(forName: "StripeSdk") as? StripeSdk
        stripeSdk?.cardFieldView = cardField;
        return cardField
    }
    
    @objc func focus(_ abi44_0_0ReactTag: NSNumber) {
        self.bridge!.uiManager.addUIBlock { (_: ABI44_0_0RCTUIManager?, viewRegistry: [NSNumber: UIView]?) in
            let view: CardFieldView = (viewRegistry![abi44_0_0ReactTag] as? CardFieldView)!
            view.focus()
        }
    }
    
    @objc func blur(_ abi44_0_0ReactTag: NSNumber) {
        self.bridge!.uiManager.addUIBlock { (_: ABI44_0_0RCTUIManager?, viewRegistry: [NSNumber: UIView]?) in
            let view: CardFieldView = (viewRegistry![abi44_0_0ReactTag] as? CardFieldView)!
            view.blur()
        }
    }
    
    @objc func clear(_ abi44_0_0ReactTag: NSNumber) {
        self.bridge!.uiManager.addUIBlock { (_: ABI44_0_0RCTUIManager?, viewRegistry: [NSNumber: UIView]?) in
            let view: CardFieldView = (viewRegistry![abi44_0_0ReactTag] as? CardFieldView)!
            view.clear()
        }
    }
    
    override class func requiresMainQueueSetup() -> Bool {
        return false
    }
}
